cli/errors.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Basic error handling for number of arguments (update call)
+func callUpdate(arg0 string, arg1 string) {
+	if arg0 != "" && arg1 != "" && len(arg1) > 5 && arg1[len(arg1)-5:] == ".json" {
+		updateUserInfo(arg0, arg1)
```

Comment:
It shouldn't be the `validator`'s responsibility to call the actual command `handler` (here, `validator` = `validateUpdate()` and `handler` = `updateUserInfo()`). The validator should simply return whether or not the syntax is `valid`, and maybe an error message that gets sent back upstream.

Put another way, it doesn't make sense for the handling of the `write` command to go _through_ the `initial validation` stage; we could instead complete `initial validation` **and then** `handle the writing process`. Having the validator call the handler makes it so that if the handler fails (ex. when actually writing information to Vault) the stack trace will include the validator as a "coordinate" in "the location where the error occurred", even though the input validation had nothing to do with the error that occurred.